frontend/src/tests/posting-tests.js

```diff
@@ -1,39 +1,356 @@
+        resp = await apiGET(`/admin/postings/${posting.id}/survey`);
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(surveyPropTypes, resp.payload);
+    });
+
+    it("Survey for a posting includes questions related to each PostingPosition", async () => {
+        resp = await apiGET(`/admin/postings/${posting.id}/survey`);
+        expect(resp).toHaveStatus("success");
+        const survey = resp.payload;
+        const postingPosQuestions = survey.pages.filter(
+            (p) => p.name === "preferences_page"
+        )[0].elements[0].rows;
+        resp = await apiGET(`/admin/postings/${posting.id}/posting_positions`);
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload.length).toEqual(postingPosQuestions.length);
+    });
+
+    it.skip("Cannot create a posting_position with a position associated with a different session than the posting", async () => {
+        const newPosting = {
+            name: "CSC500F TA",
+            intro_text: "Posting for CSC500F",
+            open_date: "2021/04/01",
+            close_date: "2021/05/01",
+            availability: "auto",
+        };
+
+        resp = await apiPOST(
+            `/admin/sessions/${session2.id}/postings`,
+            newPosting
+        );
+        expect(resp).toHaveStatus("success");
+        Object.assign(newPosting, resp.payload);
+        const newPostingPos = {
+            hours: 20,
+            num_positions: 1,
+            position_id: position.id,
+        };
+        resp = await apiPOST(
+            `/admin/postings/${newPosting.id}/posting_positions`,
+            newPostingPos
+        );
+        expect(resp).toHaveStatus("error");
+    });
+
+    // MUST run "Create a posting_position for a posting" first to load data into ‘postingPos‘
+    it("Modify a posting_position", async () => {
+        const updatePostingPos = {
+            position_id: postingPos.position_id,
+            hours: 10,
+            num_positions: 2,
+        };
+
+        resp = await apiPOST(
+            `/admin/postings/${posting.id}/posting_positions`,
+            updatePostingPos
+        );
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload).toMatchObject(updatePostingPos);
+    });
+
+    it("Cannot create two posting_positions with the same position_id for a single posting", async () => {
+        const newPostingPos = {
+            position_id: position.id,
+            hours: 10,
+            num_positions: 2,
+        };
+
+        resp = await apiPOST(
+            `/admin/postings/${posting.id}/posting_positions`,
+            newPostingPos
+        );
+        expect(resp).toHaveStatus("success");
+        resp = await apiGET(`/admin/postings/${posting.id}/posting_positions`);
+
+        // The previous POST request should not create a new posting_position and instead it upserts to an existing one
+        expect(
+            resp.payload.filter((p) => p.position_id === position.id).length
```

Comment:
`.find` instead of `.filter`

Reply:
Here I want to check explicitly that there is only one `posting_position` that has `.position_id === position.id`. Using `.find` can only make sure there exists such one but not guarantee it's the only one, right?

Reply:
That's right. `.find` will only find the first one